### PR TITLE
feat(runtime): set up a tracing-subscriber as a default feature

### PR DESCRIPTION
**Pull request by orhun**
Example log:

cap

### CODE REVIEW

**Comment by jonaro00** [comment]
The warning could say "Shuttle's default tracing subscriber is initialized!"

File: codegen/src/shuttle_main/mod.rs [hunk 1]
```diff
@@ -13,11 +13,37 @@ pub(crate) fn r#impl(_attr: TokenStream, item: TokenStream) -> TokenStream {
...
+        quote! {
+            #[tokio::main]
+            async fn main() {
+                shuttle_runtime::start(loader).await;
+            }
+        }
+    };
+
+    quote! {
+        #main
```
Comment (chesedo) [inline codegen/src/shuttle_main/mod.rs:46]:
```suggestion
    let tracing_setup = if cfg!(feature = "setup-tracing") {
        quote! {
                use shuttle_runtime::colored::*;
                shuttle_runtime::colored::control::set_override(true);
                shuttle_runtime::tracing_subscriber::fmt::init();
                println!(
                    "{}\n{}\nTo disable tracing, remove the default features from {}:\n{}\n{}",
                    "tracing is initialized!".yellow().bold(),
                    "=".repeat(23).yellow(),
                    "shuttle-runtime".italic(),
                    "shuttle-runtime = { version = \"*\", default-features = false }"
                        .white()
                        .italic(),
                    "=".repeat(23).yellow()
                );
        }
    } else {
        None
    };

    quote! {
        #[tokio::main]
        async fn main() {
            #tracing_setup
            shuttle_runtime::start(loader).await;
```

**Review by chesedo — COMMENTED** [review]
Looks good! One suggestion

**Comment by orhun** [comment]
Addressed the review comments.

**Review by jonaro00 — APPROVED** [review]

**Review by oddgrd — APPROVED** [review]
Looks great, thanks!

**Pull request merged**